### DISCUSSION TITLE
feat: add listen hook in node-server preset

### DISCUSSION THREAD

**Discussion by lewebsimple** [nitrojs/nitro#270] [Ideas]
In the context of Nuxt 3, it's easy to access the server instance using the `listen` hook in development, but in production it's quite hard (and it seems overkill) to override the nitropack `node-preset` just to hook stuff to the http server instance like WebSocket.

Is there any issue with exposing this hook ?

**Comment by stafyniaksacha** [comment]
Hi @lewebsimple 

The `listen` hooks does not raise when nitro server is listening, but when dev server is started ([see here](https://github.com/nuxt/framework/blob/main/packages/nuxi/src/commands/dev.ts#L92))

The server used by nitro in development is served with a unix socket (on unix) or on another sub port (your dev server doesn't restart when reloading, whereas this will) ([see here](https://github.com/unjs/nitro/blob/main/src/runtime/entries/nitro-dev.ts#L12-L27))

Also note that in the `listen` event you will not be able to interact with nuxtApp neither with nitroApp

I've made a simple POC to listen to websockets on nuxt 3 here: https://github.com/stafyniaksacha/nuxt-poc-websockets
It uses a custom nitro node preset to emit an event that can then be handled in a nitro plugin. Maybe this event can be inserted in default preset, but it won't work for dev mode. (that's why I create a new Server instance in dev mode) ([see here](https://github.com/stafyniaksacha/nuxt-poc-websockets/blob/main/server/plugins/websocket.ts))
